azurelinuxagent/common/osutil/default.py

```diff
@@ -1018,15 +1022,18 @@ def get_ip4_addr(self):
         return self.get_first_if()[1]
 
     def set_route_for_dhcp_broadcast(self, ifname):
-        route_cmd = "ip route add"
-        return shellutil.run("{0} 255.255.255.255 dev {1}".format(
-            route_cmd, ifname),
-                             chk_err=False)
+        try:
+            route_cmd = ["ip", "route", "add", "255.255.255.255", "dev", ifname]
+            return shellutil.run_command(route_cmd)
```

Comment:
Why is this method even returning? The callers do nothing with the return value.

Reply:
yes, i noticed that but won't fix it now. there are several implementations of the method

Reply:
It's an easy change since none of the callers do anything with the return value, but I won't block your PR for it, up to you.